d2l-button-subtle.html

```diff
@@ -83,8 +96,9 @@
 			formtarget$=[[formtarget]]
 			name$=[[name]]
 			type$=[[type]]>
-			<d2l-icon hidden=[[!icon]] icon=[[icon]] class="d2l-button-subtle-icon"></d2l-icon>
+			<d2l-icon hidden$="[[!_iconLeft(iconRight, icon)]]" icon=[[icon]] class="d2l-button-subtle-icon d2l-button-subtle-icon-left"></d2l-icon>
```

Comment:
Seems like the additional helper in JS could be avoided with by using:
```
<d2l-icon hidden$="_iconRight(...)" ...></d2l-icon>
<span>...</span>
<d2l-icon hidden$="!_iconRight(...)" ...></d2l-icon>
```

Or, even better, perhaps the JS computed properties can be avoided altogether with CSS (might need to adjust order or specificity to make this work):
```
:host() .d2l-button-subtle-icon-left,
:host([icon-right]) .d2l-button-subtle-icon-right {
   display: inline-block;
}
:host([icon-right]) .d2l-button-subtle-icon-left,
:host() .d2l-button-subtle-icon-right {
   display: none;
}
```

Reply:
To be equivalent to your JS helper, might need extra attr:
```
:host([icon-right]) [icon].d2l-button-subtle-icon-right {
```

Reply:
Yeah, I definitely like using the attribute selector to show/hide! :)

I'm trying to think of a way we could end up with just a single `<d2l-icon>` and just position it to the right or left of the text using CSS... so far I don't have any amazing ideas.

Reply:
Yeah the single `d2l-icon` would be ideal.  Might need to create the icon and append in `attached`.

Reply:
Yeah a little trickier positioning, but that might work.

Reply:
So the issue with just using `_iconRight` instead of `_iconRight` and `_iconLeft`, is that there is a case for no icon, so even though `_iconRight` is false that doesn't mean that the left icon can be shown.

Agreed that a single `d2l-icon` would be better, so I'll work on these ideas, thanks!